lib/sidebar.rb

```diff
@@ -29,185 +29,193 @@ def pre_render(page)
     end
 
     def post_write(site)
+      files = parse_files(site)
+      write_sidebar_to_files(files)
+    end
+
+    private
+
+    def parse_files(site)
       files = []
-      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
+      destination = site.config['destination']
+      Dir.glob("#{destination}/**/*.html") do |filename|
         doc = File.open(filename) { |f| Nokogiri::HTML(f) }
-        files.push(doc)
+        sanitized_filename = filename.sanitized
+
+        files.push(
+          {
+            filename: filename,
+            sanitized_filename: sanitized_filename,
+            doc: doc
+          }
+        )
 
         headers = []
-        doc.xpath('//h2 ').each do |header|
+
+        doc.xpath('//h2').each do |header|
           next unless header['id']
 
-          child = header.last_element_child
+          child_markup = header.last_element_child
           header = {
             id: header['id'],
             title: header.content.strip,
-            hash: (child['href']).to_s
+            hash: (child_markup['href']).to_s
           }
           headers.push(header)
         end
-        sanitized_filename = sanitize_filename(filename)
-        @filename_with_headers[sanitized_filename] = { headers: headers }
-      end
 
-      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
-        sanitized_filename = sanitize_filename(filename)
-        sidebar = render(sanitized_filename)
-        file = File.open(filename) { |f| Nokogiri::HTML(f) }
-        file.xpath('//*[@id="dx-sidebar-main-nav-ul"]').each do |location|
-          location.inner_html = sidebar
-        end
-        File.open(filename, 'w') { |f| f.write(file.to_html(encoding: 'UTF-8')) }
+        @pages_with_headers[sanitized_filename] = { headers: headers }
       end
 
-      # File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
+      files
     end
 
-    private
+    def write_sidebar_to_files(files)
+      files.each do |file|
+        sanitized_filename = file[:sanitized_filename]
+        filename = file[:filename]
+        doc = file[:doc]
+        sidebar_markup = nil
+
+        begin
+          sidebar_markup = render(sanitized_filename)
+        rescue StandardError => e
+          Jekyll.logger.error("           Sidebar: Unable to render sidebar_markup for '#{filename}'.")
```

Comment:
Codacy found an issue: [Line is too long. [103/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)